plugins/services/src/js/components/dsl/TaskRegionDSLSection.js

```diff
@@ -0,0 +1,81 @@
+import React, { PropTypes } from "react";
+
+import DSLCombinerTypes from "#SRC/js/constants/DSLCombinerTypes";
+import DSLExpression from "#SRC/js/structs/DSLExpression";
+import DSLExpressionPart from "#SRC/js/structs/DSLExpressionPart";
+import DSLFormWithExpressionUpdates
+  from "#SRC/js/components/DSLFormWithExpressionUpdates";
+import DSLUtil from "#SRC/js/utils/DSLUtil";
+import FieldInput from "#SRC/js/components/form/FieldInput";
+import FieldLabel from "#SRC/js/components/form/FieldLabel";
+import FormGroup from "#SRC/js/components/form/FormGroup";
+
+const EXPRESSION_PARTS = {};
+
+class TaskRegionDSLSection extends React.Component {
+  render() {
+    const { expression, onChange } = this.props;
+    const { regions } = this.props.defaultData || [];
```

Comment:
😕  I'm not sure I understand what you we're trying, but this line looks wrong to me. I'd provide a default for `defaultData` using `defaultProps` and would ensure that's an object using `propTypes` so that it's save to simply destruct  the props. Use the following code in case you want to provide a default for `regions`.

```js
 const { regions=[] } = this.props.defaultData;
```